python/ql/src/semmle/python/frameworks/Aiohttp.qll

```diff
@@ -424,24 +417,20 @@ module AiohttpWebModel {
     /**
      * Taint propagation for `aiohttp.StreamReader`.
      */
-    private class AdditionalTaintStep extends TaintTracking::AdditionalTaintStep {
-      override predicate step(DataFlow::Node nodeFrom, DataFlow::Node nodeTo) {
-        // normal (non-async) methods
-        nodeFrom = instance() and
-        nodeTo.(DataFlow::MethodCallNode).calls(nodeFrom, ["read_nowait"])
-        or
-        // async methods
-        exists(DataFlow::MethodCallNode call, Await await |
-          nodeTo.asExpr() = await and
-          nodeFrom = instance()
-        |
-          await.getValue() = any(DataFlow::Node awaitable | call.flowsTo(awaitable)).asExpr() and
-          call.calls(nodeFrom,
-            [
-              "read", "readany", "readexactly", "readline", "readchunk", "iter_chunked", "iter_any",
-              "iter_chunks"
-            ])
-        )
+    private class InstanceTaintSteps extends InstanceTaintStepsHelper {
+      InstanceTaintSteps() { this = "aiohttp.StreamReader" }
+
+      override DataFlow::Node getInstance() { result = instance() }
+
+      override string getAttributeName() { none() }
```

Comment:
I think overriding `none()` with `none()` sends the wrong signal. I think the possible methods should be plenty discoverable from the definition of `InstanceTaintStepsHelper` and from the template.

Reply:
I don't fully understand what you're suggesting here. [`getAttributeName` is defined to be abstract](https://github.com/github/codeql/blob/47377c71970ee9317172e37329035e2817558732/python/ql/lib/semmle/python/frameworks/internal/InstanceTaintStepsHelper.qll#L19-L20), so it _needs_ to be overridden.